### PR TITLE
Implemented ballInTeamHalf evaluation functions

### LINKED ISSUE #436: Implement ballInEnemyHalf function
### Description of the task

Implement a function that indicates whether or not the ball is in the enemy team's half of the field.

### Outline the task in more detail

The function should take a Field and a Ball, and return true if the ball is in the enemy's half of the field.

The function should be placed in `src/thunderbots/software/ai/hl/stp/evaluation/ball.*`, in the Evaluation namespace.

### Acceptance criteria

- [ ] Implement the function
- [ ] Unit tests

### LINKED ISSUE #437: Implement ballInFriendlyHalf function
### Description of the task

Implement a function that indicates whether or not the ball is in the friendly team's half of the field.

### Outline the task in more detail

The function should take a Field and a Ball, and return true if the ball is in the friendly team's half of the field.

The function should be placed in `src/thunderbots/software/ai/hl/stp/evaluation/ball.*`, in the Evaluation namespace.

### Acceptance criteria

- [ ] Implement the function
- [ ] Unit tests

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Implemented two evaluation functions:
1. `ballInFriendlyHalf`
2. `ballInEnemyHalf`

Both takes a Field and a Ball and returns true if the ball is in its respective half.

### Testing Done

Unit tests

### Resolved Issues

resolves #436   
resolves #437 
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
This PR finishes changes started by @jessicahe21 in #538

File: src/thunderbots/software/ai/hl/stp/evaluation/ball.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    /**
+     * Returns true if the ball is in the enemy half of the field, and false otherwise
+     *
+     * @param field The field
+     * @param ball The ball
+     * @return true if the ball is in the enemy's half of the field, and false otherwise
+     */
+    bool ballInEnemyHalf(const Field &field, const Ball &ball);
+}  // namespace Evaluation
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/evaluation/ball.h]: github complaining about end-of-file

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Just one comment, looks sane otherwise

File: src/thunderbots/software/ai/hl/stp/evaluation/ball.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/evaluation/ball.h]: fixed

**Review by garethellis0 — APPROVED** [review]

**Comment by garethellis0** [comment]
LGTM :+1: :tada:

**Pull request merged**